src/lib/core/mp_datetime.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2021, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include <assert.h>
+#include <limits.h>
+
+#include "msgpuck.h"
+#include "mp_datetime.h"
+#include "mp_extension_types.h"
+#include "mp_utils.h"
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
+	      "Tarantool currently only supports little-endian hardware.");
+
+/*
+  Datetime MessagePack serialization schema is MP_EXT extension, which
+  creates container of 8 or 16 bytes long payload.
+
+  +---------+--------+===============+-------------------------------+
+  |0xd7/0xd8|type (4)| seconds (8b) | nsec; tzoffset; tzindex; (8b)  |
+  +---------+--------+===============+-------------------------------+
+
+  MessagePack data encoded using fixext8 (0xd7) or fixext16 (0xd8), and may
+  contain:
+
+  - [required] seconds parts as full, unencoded, signed 64-bit integer, stored
+    in little-endian order;
+
+  - [optional] all the other fields (nsec, tzoffset, tzindex) if any of them
+     were having not 0 value. They are packed naturally in little-endian order;
+
+ */
```

Comment:
Do you feel the need to automate this formatting process? Isn't it annoying to ask once, and once again to convert to "dxygen" comments here and there? Just imagine that one day we could run some tool, let name it `vlad-format` and it would automagically convert all outer comments to doxygen, properly prepending every line with nice star!...

Reply:
Fixed (manually this time):
```diff
--- a/src/lib/core/mp_datetime.c
+++ b/src/lib/core/mp_datetime.c
@@ -16,23 +16,23 @@
 static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
 	      "Tarantool currently only supports little-endian hardware.");
 
-/*
-  Datetime MessagePack serialization schema is MP_EXT extension, which
-  creates container of 8 or 16 bytes long payload.
-
-  +---------+--------+===============+-------------------------------+
-  |0xd7/0xd8|type (4)| seconds (8b)  | nsec; tzoffset; tzindex; (8b) |
-  +---------+--------+===============+-------------------------------+
-
-  MessagePack data encoded using fixext8 (0xd7) or fixext16 (0xd8), and may
-  contain:
-
-  - [required] seconds parts as full, unencoded, signed 64-bit integer, stored
-    in little-endian order;
-
-  - [optional] all the other fields (nsec, tzoffset, tzindex) if any of them
-     were having not 0 value. They are packed naturally in little-endian order;
-
+/**
+ * Datetime MessagePack serialization schema is MP_EXT extension, which
+ * creates container of 8 or 16 bytes long payload.
+ *
+ * +---------+--------+===============+-------------------------------+
+ * |0xd7/0xd8|type (4)| seconds (8b)  | nsec; tzoffset; tzindex; (8b) |
+ * +---------+--------+===============+-------------------------------+
+ *
+ * MessagePack data encoded using fixext8 (0xd7) or fixext16 (0xd8), and may
+ * contain:
+ *
+ * - [required] seconds parts as full, unencoded, signed 64-bit integer, stored
+ *   in little-endian order;
+ *
+ * - [optional] all the other fields (nsec, tzoffset, tzindex) if any of them
+ *    were having not 0 value. They are packed naturally in little-endian order;
+ *
  */
 
 #define SZ_TAIL sizeof(struct datetime) - sizeof(((struct datetime *)0)->epoch)
@@ -40,8 +40,7 @@ static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
 static inline uint32_t
 mp_sizeof_datetime_raw(const struct datetime *date)
 {
-	uint32_t sz = 0;
-	sz = sizeof(int64_t);
+	uint32_t sz = sizeof(int64_t);
 	if (mp_unlikely(date->tzoffset != 0 || date->tzindex != 0 ||
 			date->nsec != 0))
 		sz += SZ_TAIL;
```